src/snowflake/snowpark/_internal/open_telemetry_prototype.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+import inspect
+from typing import Dict
+
+# this parameter make sure no error when open telemetry is not installed
+open_telemetry_not_found = False
+try:
+    from opentelemetry import trace
+    from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
+
+except ImportError:
+    open_telemetry_not_found = True
+
+if not open_telemetry_not_found:
+    resource = Resource(attributes={SERVICE_NAME: "snowpark-python-open-telemetry"})
+    # output to console for debug
+    traceProvider = TracerProvider(resource=resource)
+    processor = BatchSpanProcessor(ConsoleSpanExporter())
```

Comment:
I feel we don't need to set an exporter? I don't remember the details.

Reply:
Yes, this is only for testing.